cmd/client/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+func main() {
+	endpoint := "http://localhost:8080/"
+	// контейнер данных для запроса
+	data := url.Values{}
+	// приглашение в консоли
+	fmt.Println("Введите длинный URL")
+	// открываем потоковое чтение из консоли
+	reader := bufio.NewReader(os.Stdin)
+	// читаем строку из консоли
+	long, err := reader.ReadString('\n')
+	if err != nil {
+		panic(err)
+	}
+	long = strings.TrimSuffix(long, "\n")
+	// заполняем контейнер данными
+	data.Set("url", long)
+	// добавляем HTTP-клиент
+	client := &http.Client{}
+	// пишем запрос
+	// запрос методом POST должен, помимо заголовков, содержать тело
+	// тело должно быть источником потокового чтения io.Reader
+	request, err := http.NewRequest(http.MethodPost, endpoint, strings.NewReader(data.Encode()))
+	if err != nil {
+		panic(err)
+	}
+	// в заголовках запроса указываем кодировку
+	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	// отправляем запрос и получаем ответ
+	response, err := client.Do(request)
+	if err != nil {
+		panic(err)
+	}
+	// выводим код ответа
+	fmt.Println("Статус-код ", response.Status)
+	defer response.Body.Close()
```

Comment:
`defer response.Body.Close()` стоит вызывать сразу после обработки ошибки, потому как если при вызове промежуточного кода произойдет ошибка, Body останется не закрыт.
```suggestion
	defer response.Body.Close()
	// выводим код ответа
	fmt.Println("Статус-код ", response.Status)
```